packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart

```diff
@@ -0,0 +1,215 @@
+
+  /// Returns the credentials stored in the Keychain for the website associated with the current app.
+  ///
+  /// Only available on Apple platforms.
+  ///
+  /// Throws a [SignInWithAppleException] exception when no credentials have been found in the Keychain.
+  static Future<AuthorizationCredentialPassword> getKeychainCredential() async {
+    try {
+      if (!Platform.isIOS &&
+          !Platform.isMacOS &&
+          Platform.environment['FLUTTER_TEST'] != 'true') {
+        throw SignInWithAppleNotSupportedException(
+          message: 'The current platform is not supported',
+        );
+      }
+
+      return parseAuthorizationCredentialPassword(
+        await channel.invokeMethod<Map<dynamic, dynamic>>(
+          'performAuthorizationRequest',
+          [
+            PasswordAuthorizationRequest(),
+          ].map((request) => request.toJson()).toList(),
+        ),
+      );
+    } on PlatformException catch (exception) {
+      throw SignInWithAppleException.fromPlatformException(exception);
+    }
+  }
+
+  /// Requests an Apple ID credential.
+  ///
+  /// Shows the native UI on Apple's platform and a Chrome Custom Tab on Android.
+  ///
+  /// The returned [AuthorizationCredentialAppleID]'s `authorizationCode` should then be used to validate the token with Apple's servers and
+  /// create a session in your system.
+  ///
+  /// Fields on the returned [AuthorizationCredentialAppleID] will be set based on the given scopes.
+  ///
+  /// User data fields (first name, last name, email) will only be set if this is the initial authentication between the current app and Apple ID.
+  ///
+  /// The returned Future will resolve in all cases on iOS and macOS, either with an exception if Sign in with Apple is not available,
+  /// or as soon as the native UI goes away (either due cancellation or the completion of the authorization).
+  ///
+  /// On Android the returned Future will never resolve in case the user closes the Chrome Custom Tab without finsihing the authentication flow.
+  /// Any previous Future would be rejected if the [getAppleIDCredential] is called again, while an earlier one is still pending.
+  ///
+  /// Throws an [SignInWithAppleException] in case there was any error retrieving the credential.
+  /// A specialized [SignInWithAppleAuthorizationException] is thrown in case of authorization errors, which contains
+  /// further information about the failure.
+  ///
+  /// Throws an [SignInWithAppleNotSupportedException] in case Sign in with Apple is not available (e.g. iOS < 13, macOS < 10.15)
+  static Future<AuthorizationCredentialAppleID> getAppleIDCredential({
+    @required List<AppleIDAuthorizationScopes> scopes,
+
+    /// Optional parameters for web-based authentication flows on non-Apple platforms
+    ///
+    /// This parameter is required on Android.
+    WebAuthenticationOptions webAuthenticationOptions,
+  }) async {
+    assert(scopes != null);
+
+    if (webAuthenticationOptions == null &&
```

Comment:
should we move this check into the `if (Platform.isAndroid) {` potentially?

Reply:
Done